core/domain/acl_decorators.py

```diff
@@ -343,24 +343,89 @@ def test_can_access(self, **kwargs):
     return test_can_access
 
 
-def can_comment_on_feedback_thread(handler):
-    """Decorator to check whether the user can view feedback for a given
-    exploration.
+def can_create_feedback_thread(handler):
```

Comment:
Will this need to be generalized in the future as can_create_exploration_feedback_thread?

Thinking about what happens when we have feedback threads for e.g. skills. The name suggests that we would use this decorator there as well, but the implementation suggests otherwise.

Or perhaps we keep it as can_create_feedback_thread but pass in info related to the thread type etc.? I think that would probably be cleaner, but it depends on whether it's reasonable to represent "can_create_feedback_thread" as a single action type across the board. I think it is (i.e. you shouldn't need different types of permissions for creating feedback threads on different things).

Reply:
With the generalised version, we will need 2 parameters -- entity type and entity id. If the user can access(play) the  entity, the user should be able to create a feedback thread. Based on the entity type, I think the appropriate 'can access entity' or similar function needs to be called.

Reply:
I'm fine with that; the granularity of the actions should match the granularity needed by the different roles.